python/triton/compiler/compiler.py

```diff
@@ -163,6 +163,7 @@ def compile(src, target=None, options=None):
     if not isinstance(src, ASTSource):
         assert isinstance(src, str), "source must be either AST or a filepath"
         src = IRSource(src)
+    name = str(src.fn) if isinstance(src, ASTSource) else src.path
```

Comment:
Is it possible to parse the IR and get the kernel name?

Reply:
this would not be unique though

Reply:
I see. My concern is just that the variable `name` sound more like the name of the kernel.

Reply:
that's fair. Especially we are using str(src.fn) in the ASTSource case, and IIRC this is not unique

Reply:
suggestion to name it as compiled_kernel_name

Reply:
@Jokeren the problem is that we need a name that is available at the JITFunction level and at the CompiledKernel level. For the IRSource case, we just use the path, which is basically the old behavior before the refactor.

I agree, I can rename the variable to something else.